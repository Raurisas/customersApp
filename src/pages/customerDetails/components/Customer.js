import React from 'react';
import Button from '../../../shared/components/Button';

const Customer = ({ customer, backClickHandler }) => {
  return (
    <div className="customer">
      <Button onClick={backClickHandler} label="Go back" type="button" />
      <div className="customer-details">
        <div>Email: {customer.email}</div>
        <div>Full name:{customer.metadata.fullName}</div>
        <div>City: {customer.metadata.city}</div>
        <div>Street: {customer.metadata.street}</div>
        <div>House number:{customer.metadata.houseNumber}</div>
        <div>Zipcode: {customer.metadata.zipCode}</div>
      </div>
    </div>
  );
};

export default Customer;
