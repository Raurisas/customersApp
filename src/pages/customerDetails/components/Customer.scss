.customer {
  margin: 0 auto;
  .customer-details {
    border: 1px solid $border-color;
    display: flex;
    justify-content: center;
    flex-direction: column;
    margin: 0 auto;
    padding: 10px 30px;
    margin-top: 10px;
    & > div {
      margin: 10px 0px;
    }
  }
}

