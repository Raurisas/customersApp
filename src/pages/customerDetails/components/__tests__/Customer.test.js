import React from 'react';
import { shallow } from 'enzyme';
import Customer from '../Customer';

it('should match snapshot', () => {
  const customer = {
    email: 'aurimas993@gmail.com',
    metadata: {
      fullName: 'Aurimas Jackevicius',
      city: 'Vilnius',
      street: 'Ulvydo',
      houseNumber: '7',
      zipCode: '08250'
    }
  };

  const wrapper = shallow(<Customer customer={customer} backClickHandler={() => {}} />);
  expect(wrapper).toMatchSnapshot();
});
