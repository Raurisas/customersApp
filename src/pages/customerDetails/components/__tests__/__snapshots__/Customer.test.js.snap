// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="customer"
>
  <Button
    label="Go back"
    onClick={[Function]}
    type="button"
  />
  <div
    className="customer-details"
  >
    <div>
      Email: 
      aurimas993@gmail.com
    </div>
    <div>
      Full name:
      Aurimas Jackevicius
    </div>
    <div>
      City: 
      Vilnius
    </div>
    <div>
      Street: 
      Ulvydo
    </div>
    <div>
      House number:
      7
    </div>
    <div>
      Zipcode: 
      08250
    </div>
  </div>
</div>
`;
